### DISCUSSION TITLE
About how to set a custom path for local_file_picker in the example

### DISCUSSION THREAD

**Discussion by levywang** [zauberzeug/nicegui#2863] [Q&A] [answered]
### Question

Dear developers, as mentioned in the title, I want to make some modifications based on the local_file_picker in the example, such as setting the default opening path to the /data/users directory. How should I do this? Thank you for your help

**Comment by levywang** [comment]
Sorry, I found a way to add the default path, just add the directory path in the first parameter of the function.

```py
async def pick_file() -> None:
    result = await local_file_picker('/data/users/', multiple=True)
    ui.notify(f'You chose {result}')
```

But I still have a question, I hope you can help me answer it, how to trigger the pick_file function without using the onclick method of ui.button?

**Reply by falkoschindler** [reply]
Hi @levywang, you could call `pick_file` when there's a new client connection:
```py
#!/usr/bin/env python3
from local_file_picker import local_file_picker
from nicegui import ui, app

async def pick_file() -> None:
    result = await local_file_picker('/data/users/', multiple=True)
    ui.notify(f'You chose {result}')

app.on_connect(pick_file)

ui.run()
```

**Reply by levywang** [reply]
Hi @falkoschindler 

There is still a problem. When I use app.on_connect to trigger the pick_file method, it will bypass my auth module and directly pop up the function in pick_file on any page. Obviously this is wrong. How can I correct it? Thank you. your answer.

I used the authentication authentication method in the example

[https://github.com/zauberzeug/nicegui/blob/main/examples/authentication/main.py](https://github.com/zauberzeug/nicegui/blob/main/examples/authentication/main.py)

**Reply by falkoschindler** [reply, marked as answer]
When using individual pages, you can await the connection of the current client like this:
```py
from local_file_picker import local_file_picker
from nicegui import ui, Client

async def pick_file() -> None:
    result = await local_file_picker('/data/users/', multiple=True)
    ui.notify(f'You chose {result}')

@ui.page('/')
async def page(client: Client):
    await client.connected()
    await pick_file()

ui.run()
```

**Reply by levywang** [reply]
Now it works fine, thank you very much
